cicecore/cicedynB/dynamics/ice_dyn_evp.F90

```diff
@@ -1203,7 +1203,114 @@ subroutine stress_T   (nx_block,   ny_block,   &
       endif
 
     end subroutine stress_T
+
+    !=======================================================================
+
+! Computes divergence of stress tensor at the E or N point for the mom equation
+      
+! author: JF Lemieux, ECCC
+! Nov 2021      
+
+      subroutine div_stress  (nx_block,   ny_block,   & 
+                              ksub,       icell,     & 
+                              indxi,     indxj,     &
+                              dxE_N,   dyE_N, &
+                              dxT_U,   dyT_U, &
+                              arear,     &
+                              stressp,   stressm,     & 
+                              stress12, &
+                              F1, F2,               &
+                              grid_location)
+
+        use ice_dyn_shared, only: strain_rates_T, deformations_T, &
+                                  viscous_coeffs_and_rep_pressure_T
+        
+      integer (kind=int_kind), intent(in) :: & 
+         nx_block, ny_block, & ! block dimensions
+         ksub              , & ! subcycling step
+         icell                 ! no. of cells where epm (or npm) = 1 
+
+      integer (kind=int_kind), dimension (nx_block*ny_block), intent(in) :: &
+         indxi   , & ! compressed index in i-direction
+         indxj       ! compressed index in j-direction
+
+
+      real (kind=dbl_kind), dimension (nx_block,ny_block), intent(in) :: &
+         dxE_N , & ! width of E or N-cell through the middle (m)
+         dyE_N , & ! height of E or N-cell through the middle (m)
+         dxT_U , & ! width of T or U-cell through the middle (m)
+         dyT_U , & ! height of T or U-cell through the middle (m)
+         arear    , & ! 1/earea or 1/narea
+         stressp  , & ! sigma11+sigma22
+         stressm  , & ! sigma11-sigma22
+         stress12     ! sigma12
+
+      character(len=*), intent(in) :: &
+         grid_location ! E (East) or N (North) ! TO BE IMPROVED!!!!
+      
+      real (kind=dbl_kind), dimension (nx_block,ny_block), intent(out) :: &
+         F1      , & ! div of stress tensor for u component
+         F2          ! div of stress tensor for v component
```

Comment:
@JFLemieux73  @apcraig 

I  just noticed the `intent(out)` instead of  `intent(inout)`. To my knowledge it's safer to use `intent(inout)` here because we are not setting the whole array, just the indices in `icell`. An according to the Fortran standard, if I remember correctly the compiler is free to do anything with the array values at indices that are *not* explicitely set, so any zero-initialization we do is not guaranteed to still be zero after the call to this subroutine...